src/pkg/blob/dao/dao.go

```diff
@@ -162,13 +177,46 @@ func (d *dao) GetBlobByDigest(ctx context.Context, digest string) (*models.Blob,
 	return blob, nil
 }
 
-func (d *dao) UpdateBlob(ctx context.Context, blob *models.Blob) error {
+func (d *dao) UpdateBlobStatus(ctx context.Context, blob *models.Blob) error {
 	o, err := orm.FromContext(ctx)
 	if err != nil {
 		return err
 	}
 
-	_, err = o.Update(blob)
+	// each update will auto increase version and update time
+	data := make(beego_orm.Params)
+	data["version"] = beego_orm.ColValue(beego_orm.ColAdd, 1)
+	data["update_time"] = time.Now()
+	data["status"] = blob.Status
+
+	qt := o.QueryTable(&models.Blob{})
+	cond := beego_orm.NewCondition()
+	c := cond.And("version__gte", blob.Version)
+
+	/*
+		generated simple sql string.
+		UPDATE "blob" SET "version" = "version" + $1, "update_time" = $2, "status" = $3
+		WHERE "id" IN ( SELECT T0."id" FROM "blob" T0 WHERE T0."version" >= $4 AND T0."id" = $5 AND T0."status"  IN ('delete', 'deleting')  )
```

Comment:
UPDATE "blob" SET version=version+1, update_time=?, status=? WHERE id = ? AND status in (?, ?) AND version >= ?

Reply:
designed by beego orm